GitUI/CommandsDialogs/BrowseDialog/FormCommitCount.cs

```diff
@@ -28,69 +34,98 @@ private void cbIncludeSubmodules_CheckedChanged(object sender, EventArgs e)
 
         private void FetchData()
         {
-            Loading.Visible = true;
-
-            CommitCount.Text = "";
-            var dict = new Dictionary<string, HashSet<string>>();
-            var (items, _) = CommitCounter.GroupAllCommitsByContributor(Module);
-            if (cbIncludeSubmodules.Checked)
-            {
-                var submodules = Module.GetSubmodulesLocalPaths();
+            _backgroundLoaderTokenSource.Cancel();
+            _backgroundLoaderTokenSource = new CancellationTokenSource();
+            var token = _backgroundLoaderTokenSource.Token;
 
-                foreach (var submoduleName in submodules)
+            ThreadHelper.JoinableTaskFactory.RunAsync(
+                async () =>
                 {
-                    GitModule submodule = Module.GetSubmodule(submoduleName);
-                    if (submodule.IsValidGitWorkingDir())
+                    Loading.Visible = true;
+
+                    CommitCount.Text = "";
+
+                    var includeSubmodules = cbIncludeSubmodules.Checked;
+
+                    await TaskScheduler.Default.SwitchTo(alwaysYield: true);
```

Comment:
:bulb: This should probably be:

```csharp
await TaskScheduler.Default;
```

The use of `alwaysYield: true` is a defensive programming technique where I intentionally preserved the behavior of existing code, even though it is generally sub-optimal.

Reply:
Makes sense. Thanks.